src/index.rs

```diff
@@ -24,9 +24,7 @@ pub struct IndexEntry {
 
 #[derive(Clone, Debug)]
 pub struct ResourceIndex {
-    pub id2path: HashMap<ResourceId, CanonicalPathBuf>,
```

Comment:
Probably we should keep it but make it private. The `id2path` mapping removes necessity to create a cache for the `find(|(_, v)| v.id == *id)` computation. The trade-off is that the update logic is more complicated, but the `main` version looks more performant.

Reply:
Fair enough. Found it a bit confusing for what is happening as they both have the similar names but the collections get out of sync quickly.

How to check performance?

Also I forgot to run cargo fmt, which is why the build failed :/

Reply:
> Found it a bit confusing for what is happening as they both have the similar names but the collections get out of sync quickly.

Probably we should introduce some `fn check_consistency()` and use it in unit tests.

> How to check performance?

Oh yeah, we need some benchmarks..